draft-ietf-tls-wkech.xml

```diff
@@ -345,120 +342,147 @@ Client <-->|                    | |                    | facing
-            </t>
-            <t>
-                Origins that simply alias to a single target can indicate this
-                without copying the ECHConfig and other parameters, which
-                can interfere with key rotation and other maintenance.
-            </t>
-            <t>
-                "port" and "target" are generally sufficient to uniquely identify
-                a ServiceMode record, so zone factories can use the endpoint list
-                to add ECH to pre-existing ServiceMode records that may have other
-                SvcParams.
-            </t>
-        </list>
+    <t>The following keys are defined for ServiceMode entries:
+    <list style="symbols">
+      <t>
+	  target: The value is a string containing a fully qualified
+	  domain name, corresponding to the HTTPS record's TargetName.
+	  The default value is ".".  All labels in the name MUST contain
+	  only octets from the ABNF set ALPHA / DIGIT / "-" / "_".
+      </t>
+      <t>
+	  priority: The value is a positive integer corresponding to the
+	  SvcPriority.  If omitted, the ZF MAY infer
+	  numerically increasing SvcPriority from the order of the
+	  endpoints array.
+      </t>
+      <t>
+	  params: A JSON Dictionary representing the SVCB SvcParams.  Each
+	  key in the dictionary is a string containing a registered
+	  SvcParamKey name (e.g., "ipv6hint") or a SvcParamKey in generic
+	  form (e.g., "key65528").
+	<list>
+	  <t>
+	      For single-valued SvcParams (e.g., "ech"), the value is a JSON
+	      String.  A JSON String is a sequence of Unicode codepoints,
+	      while a SvcParam's presentation value is a sequence of octets,
+	      so each value octet is stored as a single Unicode codepoint
+	      <xref target="ISOMORPHIC-DECODE"/>.  In almost all cases, this
+	      is equivalent to the ordinary string representation of the
+	      presentation value.
+	  </t>
+	  <t>
+	      For list-valued SvcParams (e.g., "alpn"), the value is a JSON Array
+	      of Strings.  Each String represents an octet sequence, as in the
+	      single-value case.
+	  </t>
+	</list>
+      </t>
+    </list>
+  </t>
+
+    <t>The following key is defined for AliasMode entries.
+
+    <list style="symbols">
+      <t>
+	  alias: The value MUST be a DNS name that could be used as the
+	  TargetName of an HTTPS resource record.  This indicates that the
+	  backend is hosted on the same endpoints as this target, and is
+	  equivalent to an HTTPS AliasMode record.  The ZF might implement
+	  this directive by publishing an AliasMode record, publishing a
+	  CNAME record, copying HTTPS records from the target zone, or
+	  fetching https://cfs.example.com/.well-known/origin-svcb (if it
+	  exists).  In this case, the regeninterval indicates that backend
+	  does not plan to change the content at the URL for at least that
+	  number of seconds.  If an alias entry is present then any ech and
+	  port entries (if also present) MUST be ignored.
```

Comment:
```suggestion
	  exists).
```

I think "regeninterval" should consistently mean "how often this JSON file is updated", and the backend can make its own decisions about how often to update its configuration.

I don't think the "ech" and "port" rule applies anymore.